Features/search bkend 84

dan-crayne opened the pull request:
Sorry for the confusing commit history; I had made some rookies mistakes when merging while I was still in process.

dan-crayne committed:
[#84] remove linkedin_id change record descriptions

The linkedin_id field is no longer needed, and probably will be removed.

Changed the descriptions to something that contains more English words
so that the text will be more friendly to text searches than the Lorem
Ipsum words.

dan-crayne committed:
[#84] remove linkedin_id change record descriptions

The linkedin_id field is no longer needed, and probably will be removed.

Changed the descriptions to something that contains more English words
so that the text will be more friendly to text searches than the Lorem
Ipsum words.

dan-crayne committed:
[#84] remove linkedin_id change record descriptions

The linkedin_id field is no longer needed, and probably will be removed.

Changed the descriptions to something that contains more English words
so that the text will be more friendly to text searches than the Lorem
Ipsum words.

dan-crayne committed:
add simple pattern search

This simple search returns a json result based on a pattern match of a
search term given as a query string parameter:
e.g. http://localhost:5000/search?term='club'

It still needs the results to be added to a single 'results' root,
instead of each record type (user,org,prog) receiving their own 'result'
root.

dan-crayne committed:
add simple pattern search

This simple search returns a json result based on a pattern match of a
search term given as a query string parameter:
e.g. http://localhost:5000/search?term='club'

It still needs the results to be added to a single 'results' root,
instead of each record type (user,org,prog) receiving their own 'result'
root.

dan-crayne committed:
add simple pattern search

This simple search returns a json result based on a pattern match of a
search term given as a query string parameter:
e.g. http://localhost:5000/search?term='club'

It still needs the results to be added to a single 'results' root,
instead of each record type (user,org,prog) receiving their own 'result'
root.

dan-crayne committed:
add simple pattern search

This simple search returns a json result based on a pattern match of a
search term given as a query string parameter:
e.g. http://localhost:5000/search?term='club'

It still needs the results to be added to a single 'results' root,
instead of each record type (user,org,prog) receiving their own 'result'
root.

dan-crayne committed:
refactor - removed where clause builder

dan-crayne committed:
refactor - removed where clause builder

dan-crayne committed:
refactor - removed where clause builder

dan-crayne committed:
refactor - removed where clause builder

dan-crayne committed:
Merge branch 'features/search_backend_84' of https://github.com/iandusenbury/tableland into features/search_backend_84

need serialization features to continue search changes

dan-crayne committed:
remove where clause builder

dan-crayne committed:
remove where clause builder

dan-crayne committed:
add regular admin for testing

dan-crayne committed:
move base controller to 'v1' scope and replace calls to search

dan-crayne committed:
remove text left over from merge

dan-crayne committed:
remove extra search method

dan-crayne committed:
reconcile for merge

dan-crayne committed:
reconcile for merge

dan-crayne committed:
add search class for model common search tasks

dan-crayne committed:
flatten results json

If the results are not flattened, the serializer is not able to process
them.

dan-crayne committed:
add search to non-authenticated routes

dan-crayne committed:
flatten results before rendering

The serializers are not able to process the results if they are not
flattened first.

dan-crayne committed:
move where clause generation to common class

dan-crayne committed:
refactor search calls

dan-crayne committed:
merge with development

dan-crayne committed:
add experience titles to user search

dan-crayne committed:
tune search fields for more relevant results

Note (search fields excluded):
  We have chosen to not search record descriptions, as it creates confusing
  results. When a user searches for a term, often that term will appear in
  the middle of a sentence or as part of a word; this is helpful in a
  organization name or user title, but not when searching a potentially
  long bio or organization description.

Note (search fields included):
  I think it makes sense to leave out the url as a search field, because the
  url may not have anything to do with the organization or program, and I don't
  think that many people will be trying to search by url.

Note (tying users to experiences):
  The experiences of a particular user will be searched in their historical
  entirety, and results will be returned for any experience matching the
  pattern. This is because, for example, someone with a current title of
  'project manager', was probably a 'junior developer' or something similar
  in the past, and we want the users to be able to see those profiles; the
  profile's current title will appear in the results, but the user can then
  click on that result to get their previous occupations.

dan-crayne committed:
change pattern matching to be 'starts with' instead of 'contains'

dan-crayne committed:
make seed data more focused on STEM careers

evwhite29 commented on app/controllers/application_controller.rb:
```diff
@@ -1,6 +1,6 @@
 class ApplicationController < ActionController::API
   # https://github.com/gonzalo-bulnes/simple_token_authentication#make-models-token-authenticatable
-  acts_as_token_authentication_handler_for User, except: [:search, :show, :linkedin, :random], fallback: :exception
+  acts_as_token_authentication_handler_for User, except: [:search, :index, :show, :linkedin], fallback: :exception
```
I don't agree with this change. The whitelist of publicly available endpoints (actions) was cemented in a previous branch and should not be altered.

dan-crayne replied:
I didn't intend to make that change. I think this is a relic of my code being out of sync with the main branch.

dan-crayne committed:
correct code to sync with historical change to main repo

evwhite29 reviewed (dismissed):
Looks good!

dan-crayne committed:
change search selection to only include visible records; make example seeds

Seeds had previously not included invisible records since 'visible' cannot
be set during record creation (all records get 'true'); so a sample
percentage of records are updated to 'visible: false' after the
collections have been generated.

iandusenbury reviewed (approved):
Looks good to me

dan-crayne committed:
refactor default visibility assignments
